index.js

```diff
@@ -23,7 +23,13 @@ if (!hyperdb.createReadStream) {
 function Graph (storage, key, opts) {
   if (!(this instanceof Graph)) return new Graph(storage, key, opts)
   events.EventEmitter.call(this)
-  this.db = hyperdb(storage, key, opts)
+  opts = opts || {}
+  if (opts.db) {
+    this.db = opts.db
```

Comment:
By having db as an opts variable, it makes the primary argument `storage` redundant. So perhaps we could achieve this instead by making storage itself optionally be a hyperdb instance. The question then is how to differentiate it from a storage function. Any thoughts?
Also Hello! I have been meaning to get in touch since returning to Australia.